frontend/src/scenes/actions/ActionsTable.tsx

```diff
@@ -46,8 +46,39 @@ export function ActionsTable(): JSX.Element {
     const { loadActions } = useActions(actionsModel)
     const [searchTerm, setSearchTerm] = useState('')
     const [filterByMe, setFilterByMe] = useState(false)
+    const [dupVisible, setDupVisible] = useState(false)
+    const [dupName, setDupName] = useState('')
+    const [dupAction, setDupAction] = useState<ActionType>()
```

Comment:
It's not necessary to use a state variable for the action, as you can just directly pass the action to the `duplicateAction` function

Reply:
@thmsobrmlr I have completed almost all the changes. But I am not sure how to implement this one. Can you guide me a little? I am calling the `duplicateAction()` in the modal component. So not sure how i can directly pass the action as an argument.